src/extension.ts

```diff
@@ -25,33 +25,52 @@ export function start(context: theia.PluginContext): void {
     mocha.useColors(true);
 
     const e = (c: any) => console.log(c);
-
     ncp(context.extensionPath, '/projects/Che-Java-Tests', (err: any) => {
         if (err) {
             return console.error(err);
         }
+        activateJavaLSPlugin().then(() => {
+            theia.workspace.findFiles('**/tests/*.test.ts', undefined).then(files => {
+                console.log("Found: ");
+                console.log(files);
 
-        theia.workspace.findFiles('**/tests/*.test.ts', undefined).then(files => {
-            console.log("Found: ");
-            console.log(files);
-
-            // Add files to the test suite
-            files.forEach(f => mocha.addFile(path.resolve(f.path)));
+                // Add files to the test suite
+                files.forEach(f => mocha.addFile(path.resolve(f.path)));
 
-            try {
-                // Run the mocha test
-                mocha.run((failures: any) => {
-                    if (failures > 0) {
-                        e(new Error(`${failures} tests failed.`));
-                    }
-                });
-            } catch (err) {
-                e(err);
-            }
+                try {
+                    // Run the mocha test
+                    mocha.run((failures: any) => {
+                        theia.window.showInformationMessage('Tests completed! See results in test.log file');
+                        const resultFile = path.resolve('/projects', 'test.log');
+                        theia.commands.executeCommand('file-search.openFile', resultFile)
+                        if (failures > 0) {
+                            e(new Error(`${failures} tests failed.`));
+                        }
+                    });
+                } catch (err) {
+                    e(err);
+                }
+            });
         });
     });
+}
 
-
+function activateJavaLSPlugin(): Promise<void> {
+    return Promise.resolve(theia.workspace.findFiles('MyHelloText.java', null, 1)).then(async file => {
```

Comment:
just a FYI, maybe code would be more readable with an await instruction on findFiles
and use throw new Error instead of 'console.error'
``typescript
const files = await theia.workspace.findFiles('MyHelloText.java', null, 1);
if (files.length != 1) {
  throw new Error('Cannot find MyHelloText java file');
}
const file = files[0];
```

Reply:
and instead of return Promise.resolve, use `async` keyword : `async function activateJavaLSPlugin(): Promise<void> {`

Reply:
thank you for advice, updated

Reply:
thx